src/NetworkPort.php

```diff
@@ -63,19 +63,19 @@ class NetworkPort extends CommonDBChild
 
     /**
      * Subset of input that will be used for NetworkPortInstantiation.
-     * @var array
+     * @var array|null
      */
-    private $input_for_instantiation;
+    private ?array $input_for_instantiation;
```

Comment:
Maybe better to prevent "access before initialization" (appplicable to other "input*" arrays):
```suggestion
    private array $input_for_instantiation = [];
```

Reply:
Some logic in `splitInputForElements` checks for null values. I don't know enough about this to know if checking for empty arrays instead would break it or something else.

Reply:
OK; I see.

As far as I can see; null are used here:
```
        if (
            $this->input_for_instantiation !== null
            || $this->input_for_NetworkName !== null
            || $this->input_for_NetworkPortConnect !== null
            || !isset($input)
        ) {
            return;
        }
        $this->input_for_instantiation      = [];
        $this->input_for_NetworkName        = [];
        $this->input_for_NetworkPortConnect = [];
```

And those lines lend to  `Typed property NetworkPort::$input_for_instantiation must not be accessed before initialization` error. So, we must replace with isset instead of null equality: `isset($this->input_for_instantiation)`; therefore we can declare property like `private array $input_for_instantiation;` (not nullable, but not instanciated).

Another issue of the same kind on line 382 ` && is_array($this->input_for_instantiation)` which can be replaced with `isset($this->input_for_instantiation)` (you can keep the `is_array` but if property is not nullable, native type will ensure this is an array if defined.